siliconcompiler/client.py

```diff
@@ -309,17 +309,25 @@ def fetch_results_request(chip):
             elif resp.status_code == 303:
                 redirect_url = resp.headers['Location']
                 can_redirect = True
-            else:
+            elif resp.status_code == 200:
                 shutil.copyfileobj(resp.raw, zipf)
-                return
+                return 0
+            elif int(resp.status_code / 100) in [4, 5]:
```

Comment:
can't you just do `400 <= resp.status_code and resp.status_code < 600` that seems much easier to read

Reply:
I guess, or maybe just `else:`